netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -607,7 +651,30 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
           logger.log(Level.WARNING, String.format("Received unexpected ping ack. "
               + "Expecting %d, got %d", p.payload(), ackPayload));
         }
-      } else {
+      }
+      else if (data.readLong() == BDP_MEASUREMENT_PING) {
+        pingreturn++;
+        data.readerIndex(0);
+        dataSizeSincePing += data.readableBytes();
+        //Calculate new window size by doubling the OBDP
+        int target = dataSizeSincePing * 2;
+        pinging = false;
+        logger.log(Level.FINER, String.format("OBDP: %d", dataSizeSincePing));
```

Comment:
This will do String.format even when not logging. Either wrap in an if to check if logging first or do `logger.log(Level.FINER, "OBDP: {0}", dataSizeSincePing)` (The {0} is because it uses `MessageFormat` for formatting by default).